pandas/plotting/_matplotlib/tools.py

```diff
@@ -288,6 +288,36 @@ def _remove_labels_from_axis(axis):
     axis.get_label().set_visible(False)
 
 
+def _has_externally_shared_axis(ax1, compare_axis):
+    """If two axes sharing an axis also have the same position, they can be referred to
+    as internally sharing the common axis (a.k.a twinning). _handle_shared_axes() is
+    only interested in axes externally sharing an axis, regardless of whether either of
+    the axes is also internally sharing with a third axes.
+
+    """
```

Comment:
Thanks for writing a docstring :) 

[Here](https://pandas.pydata.org/docs/development/contributing_docstring.html)'s the pandas guide to docstrings, if you could have a read and make this conform to the project's conventions that'd be great

Also, type annotations would help with maintainability

Reply:
Thanks for the heads up. I hope my revision is okay?

Reply:
I think so, thanks! One more suggestion from my end would be if you could document the return value, and write the function signature as
```python
def _has_externally_shared_axis(ax1, compare_axis) -> bool:
```

Reply:
Tagging @charlesdong1991 as he's done quite some work on plotting

Reply:
Done.